langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModelIT.java

```diff
@@ -456,28 +491,186 @@ void should_use_tools_with_AiService_2() {
         verifyNoMoreInteractions(calculator);
     }
 
-    static class AnniversaryDate {
-        @Tool("get the anniversary date")
-        String getCurrentDate() {
-            return "2040-03-10";
+    static class PetName {
+        @Tool("gives the name of the pet")
+        String getPetName() {
+            return "Felicette";
         }
     }
 
     @Test
     void should_support_noarg_fn() {
 
         // given
-        AnniversaryDate anniversaryDate = new AnniversaryDate();
+        PetName petName = new PetName();
 
         Assistant assistant = AiServices.builder(Assistant.class)
             .chatLanguageModel(model)
-            .tools(anniversaryDate)
+            .tools(petName)
+            .build();
+
+        // when
+        String answer = assistant.chat("What is the name of the pet?");
+
+        // then
+        assertThat(answer).contains("Felicette");
+    }
+
+    static class Transaction {
+        private final String id;
+
+        Transaction(String id) {
+            this.id = id;
+        }
+
+        @Override
+        public String toString() {
+            return "Transaction { id = '" + id + '\'' + '}';
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Transaction that = (Transaction) o;
+            return Objects.equals(id, that.id);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(id);
+        }
+    }
+
+    static class FunctionCallingService {
+        private static final Map<Transaction, String> DATASET = new HashMap<Transaction, String>() {{
+            put(new Transaction("001"), "pending");
+            put(new Transaction("002"), "approved");
+            put(new Transaction("003"), "rejected");
+        }};
+
+        @Tool("Get the status of a payment transaction identified by its transaction ID.")
+        public String paymentStatus(@P("The ID of the payment transaction") String transactionId) {
+            System.out.println("PAYMENT STATUS FOR TRANSACTION " + transactionId + " CALLED");
+            return "Transaction " + transactionId + " is " + DATASET.get(new Transaction(transactionId));
+        }
+    }
+
+    interface FunctionCallingAssistant {
+        @dev.langchain4j.service.SystemMessage("You MUST call the `paymentStatus()` function if you're asked about a payment transaction.")
+        String chat(String userMessage);
+    }
+
+    @Test
+    void should_work_with_interspersed_function_execution_results() {
```

Comment:
Could you please clarify what `interspersed` means in this context?

Reply:
It doesn't really matter here, but that was for me to remember that this could happen:
Gemini can answer with a function execution request AND a text message at the same time.
So in a way, the text message is interspersed in the function call request / response exchanges.
But it's just an extra test for function calling.